### PR TITLE
mention the hierarchical namespace for logging

### PR DESCRIPTION
**Pull request by ghoz**
Documents hierarchical  namespace log levels
https://github.com/Koenkk/zigbee2mqtt/pull/22859

### CODE REVIEW

File: docs/guide/configuration/logging.md [hunk 1]
```diff
@@ -53,9 +53,25 @@ Logging can have a significant impact on MQTT traffic. For that reason, only `in
 advanced:
   log_level: info
   log_namespaced_levels:
-    z2m:mqtt: warning
+    'z2m:mqtt': warning
 ```
 
+## Defining levels for specific namespace hierarchy
+The log levels defined in `log_namespaced_levels` will apply to the namespace itself and all namespaces below unless explicitely configured.
+In addition to the regex based debug filter [#preventing-specific-namespaces-from-being-logged] it allows to fine tune your logging needs to great detail.
```
Comment (Nerivec) [inline docs/guide/configuration/logging.md:61]: > The log levels defined in `log_namespaced_levels` will apply to the namespace itself and all namespaces below unless explicitely configured.
In addition to the regex based debug filter [#preventing-specific-namespaces-from-being-logged] it allows to fine tune your logging needs to great detail.

=>
Any log level defined in `log_namespaced_levels` will apply to the namespace itself and all namespaces below it, unless explicitly configured.

---

Regex part is described below and only applies to `debug`, so I don't think we need to mention it here.

File: docs/guide/configuration/logging.md [hunk 2]
```diff
@@ -53,9 +53,25 @@ Logging can have a significant impact on MQTT traffic. For that reason, only `in
...
+```yaml
+advanced:
+  log_level: warning
+  log_namespaced_levels:
+    'zhc': info
+    'zhc:legacy:fz' : debug
```
Comment (Nerivec) [inline docs/guide/configuration/logging.md:67]:
```yaml
advanced:
  log_level: warning
  log_namespaced_levels:
    zhc: info
    zhc:legacy:fz: debug
```

(same at the top, can remove the quotes wrapping namespace)

File: docs/guide/configuration/logging.md [hunk 3]
```diff
@@ -53,9 +53,25 @@ Logging can have a significant impact on MQTT traffic. For that reason, only `in
...
+```
+
+- All namespaces bellow `zhc` will be logged as info
+- Except all namespaces bellow `zhc:legacy:fz` that will be logged as debug
+- All the rest uses the `warning` level`
```
Comment (Nerivec) [inline docs/guide/configuration/logging.md:72]:
```markdown
- All namespaces bellow `zhc` will be logged as info
- Except all namespaces bellow `zhc:legacy:fz` that will be logged as debug
- All the rest uses the `warning` level`
```

=>

```markdown
- `zhc` and below namespaces will be logged as `info` (examples: `zhc`, `zhc:ota:common`, `zhc:legacy:tz`)
- `zhc:legacy:fz` and below namespaces will be logged as `debug` (examples: `zhc:legacy:fz`, `zhc:legacy:fz:tuya`)
- Other unspecified namespaces will use `log_level`, in this case, `warning` (examples: `z2m:mqtt`, `zh:zstack`)
```

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**